de.metas.business/src/main/java/de/metas/pricing/conditions/service/CreatePriceListSchemaRequest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.pricing.conditions.service;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.money.CurrencyConversionTypeId;
+import de.metas.pricing.conditions.PricingConditionsId;
+import de.metas.product.ProductCategoryId;
+import de.metas.product.ProductId;
+import de.metas.tax.api.TaxCategoryId;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+
+@Value
+@Builder
+public class CreatePriceListSchemaRequest
+{
+	@NonNull
+	PricingConditionsId discountSchemaId;
+
+	int seqNo;
+
+	@Nullable
+	ProductId productId;
+	@Nullable
+	ProductCategoryId productCategoryId;
+	@Nullable
+	BPartnerId bPartnerId;
+
+	@Nullable
+	TaxCategoryId taxCategoryId;
+	@NonNull
+	BigDecimal std_AddAmt;
```

Comment:
about `std_`, 'list_` and `limit_` fields.

1. i would suggest to keep them grouped, i.e. all std together, all limit together etc
2. I would improve the naming.
For example, `std_` is coming from `priceStd`, i.e. transformations and checkings related to PriceStd field
Similarly, `limit_` is related to PriceLimit and `list_` is related to PriceList.

For all those 3 categories (std, list and limit) basically you have the same fields.
It might be better to introduce a new inner class called `PriceChange` (maybe u can come up with a better name).

Reply:
Since it's not clear to me what all those fields are meant to do, i want to leave them "spread" into 3 groups instead of making an inner class. I've fixed the grouping though.